src/utils/net/http_client.ml

```diff
@@ -64,23 +63,31 @@ type request = {
     req_filter_ip : (Ip.t -> bool);
   }
 
-type content_handler = 
-  int64 -> (string * string) list -> TcpBufferedSocket.t -> int -> unit
-
 let log_prefix = "[HTTPcl]"
 
+let bracket res destroy k =
+  let x = try k res with exn -> destroy res; raise exn in
+  destroy res;
+  x
+
+let thread_pool = ThreadPool.create 4
+
 let lprintf_nl fmt =
   lprintf_nl2 log_prefix fmt
 
+let def_user_agent =
+  let ua = Curl.version_info () in
+  let v = ua.Curl.version in
+  Printf.sprintf "curl/%s" v
```

Comment:
Why not identify the real application making the request? I would use `MLDonkey/x.y.z` or, if you prefer, `MLDonkey/x.y.z curl/a.b.c`. (See example in [RFC 9110 § User-Agent](https://www.rfc-editor.org/rfc/rfc9110.html#name-user-agent))

Reply:
Also, there is a global option, used in many places:
https://github.com/ygrek/mldonkey/blob/0d4463568fa6374fec964d139a769c5966bb7599/src/daemon/common/commonOptions.ml#L1017-L1019

The value `default` uses `MLDonkey/x.y.z` format.